src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1726,37 +1744,10 @@ protected override void EmitMarshalArgumentManagedToNative()
 
                 LoadNativeArg(_ilCodeStreams.CallsiteSetupCodeStream);
             }
-
-        }
-
-        protected override void TransformNativeToManaged(ILCodeStream codeStream)
-        {
-            LoadManagedValue(codeStream);
-            LoadNativeValue(codeStream);
-            codeStream.Emit(ILOpcode.call, _ilCodeStreams.Emitter.NewToken(
-            InteropTypes.GetSafeHandle(Context).GetKnownMethod("SetHandle", null)));
-        }
-    }
-
-    class ReferenceMarshaller : Marshaller
-    {
-        protected override void AllocNativeToManaged(ILCodeStream codeStream)
-        {
-            var emitter = _ilCodeStreams.Emitter;
-            var lNull = emitter.NewCodeLabel();
-
-            // Check for null
-            LoadNativeValue(codeStream);
```

Comment:
This logic was not applicable to SafeHandle marshaling. SafeHandle marshaling allocates the handle even when the return value is null.